lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -1070,42 +1081,11 @@ spec = describe "SHELLEY_STAKE_POOLS" $ do
                 -- Make sure the rewards are not all equal:
                 rewards pool1 .> rewards pool3
 
-        it "non-myopic-rewards are based on stake" $ \ctx -> runResourceT $ do
-            eventually "rewards are smaller for smaller stakes" $ do
```

Comment:
I'm wondering if it is not due to the caching perhaps? As far as I see on mainnet, both `desirability_score` and `non_myopic_member_rewards` do not change no matter what value of `stake` parameter is provided to the endpoint.

Reply:
Can you try really big stakes on mainnet? 🤔 The `non_myopic_member_rewards` should change, especially on pools which have a low `non_myopic_member_rewards` score, when you supply a stake of `10000000000000000` (10 billion Ada).

Reply:
No, I don't see them changing even for 45 billion... e.g.

```
$ curl -X GET http://localhost:8091/v2/stake-pools?stake=1 | jq | grep -B 50 AGG1

[
  {
    "flags": [],
    "metrics": {
      "saturation": 0.0003302915840810417,
      "desirability_score": 2153289257678,
      "non_myopic_member_rewards": {
        "quantity": 2153289257678, <---------------------------------
        "unit": "lovelace"
      },
      "produced_blocks": {
        "quantity": 1,
        "unit": "block"
      },
      "relative_stake": {
        "quantity": 0,
        "unit": "percent"
      },
      "owner_stake": {
        "quantity": 15206293156,
        "unit": "lovelace"
      }
    },
    "cost": {
      "quantity": 340000000,
      "unit": "lovelace"
    },
    "margin": {
      "quantity": 0,
      "unit": "percent"
    },
    "pledge": {
      "quantity": 15200000000,
      "unit": "lovelace"
    },
    "metadata": {
      "homepage": "https://www.adagogo1.com",
      "name": "ADAgogo1",
      "ticker": "AGG1",

$ curl -X GET http://localhost:8091/v2/stake-pools?stake=45000000000000000 | jq | grep -B 50 AGG1

[
  {
    "flags": [],
    "metrics": {
      "saturation": 0.0003302915840810417,
      "desirability_score": 2153289257678,
      "non_myopic_member_rewards": {
        "quantity": 2153289257678, <---------------------------------
        "unit": "lovelace"
      },
      "produced_blocks": {
        "quantity": 1,
        "unit": "block"
      },
      "relative_stake": {
        "quantity": 0,
        "unit": "percent"
      },
      "owner_stake": {
        "quantity": 15206293156,
        "unit": "lovelace"
      }
    },
    "cost": {
      "quantity": 340000000,
      "unit": "lovelace"
    },
    "margin": {
      "quantity": 0,
      "unit": "percent"
    },
    "pledge": {
      "quantity": 15200000000,
      "unit": "lovelace"
    },
    "metadata": {
      "homepage": "https://www.adagogo1.com",
      "name": "ADAgogo1",
      "ticker": "AGG1",
```

Reply:
Ok, perhaps for smaller `non_myopic_member_rewards` they do change indeed:
```
$ curl -X GET http://localhost:8091/v2/stake-pools?stake=1 | jq | grep -B 35 GOAT4

    "metrics": {
      "saturation": 0.016319055947771045,
      "desirability_score": 42371242529,
      "non_myopic_member_rewards": {
        "quantity": 357659983, <---------------------------------
        "unit": "lovelace"
      },
      "produced_blocks": {
        "quantity": 44,
        "unit": "block"
      },
      "relative_stake": {
        "quantity": 0,
        "unit": "percent"
      },
      "owner_stake": {
        "quantity": 1075490098497,
        "unit": "lovelace"
      }
    },
    "cost": {
      "quantity": 340000000,
      "unit": "lovelace"
    },
    "margin": {
      "quantity": 2.5,
      "unit": "percent"
    },
    "pledge": {
      "quantity": 1000000000000,
      "unit": "lovelace"
    },
    "metadata": {
      "homepage": "https://goatstake.com",
      "name": "Goat Stake",
      "ticker": "GOAT4",

$ curl -X GET http://localhost:8091/v2/stake-pools?stake=45000000000000000 | jq | grep -B 35 GOAT4

    "metrics": {
      "saturation": 0.016319055947771045,
      "desirability_score": 42371242529,
      "non_myopic_member_rewards": {
        "quantity": 42371242529, <---------------------------------
        "unit": "lovelace"
      },
      "produced_blocks": {
        "quantity": 44,
        "unit": "block"
      },
      "relative_stake": {
        "quantity": 0,
        "unit": "percent"
      },
      "owner_stake": {
        "quantity": 1075490098497,
        "unit": "lovelace"
      }
    },
    "cost": {
      "quantity": 340000000,
      "unit": "lovelace"
    },
    "margin": {
      "quantity": 2.5,
      "unit": "percent"
    },
    "pledge": {
      "quantity": 1000000000000,
      "unit": "lovelace"
    },
    "metadata": {
      "homepage": "https://goatstake.com",
      "name": "Goat Stake",
      "ticker": "GOAT4",
```

Reply:
Yup. In fact, the definition of `sigma'` in the computation of `optimalRewards`,

https://github.com/input-output-hk/cardano-wallet/blob/85be73804bb6a243a8a8f73e13d144320c17dc46/lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs#L364-L384

implies that we should actually *expect* the non-myopic rewards to *not* change with the user-supplied stake for the `nOpt` pools that rank best on desirability score — the non-myopic reward prediction assumes that they will each attract an equal share `z0 = 1/nOpt` of the stake, and adding more stake to them will *not* increase the rewards at all, as `sigma'` will not increase beyond `z0`.

If we want to replicate the concept of "smaller pool" in the integration test, I think we should set the optimum number of pools to `2` in the genesis file (or any number that is small than the total number of pools that stays active in the local cluster.)

Reply:
Ok, I understand that this change was introduced in this PR to comply with delegation design spec? 

>If we want to replicate the concept of "smaller pool" in the integration test, I think we should set the optimum number of pools to 2 in the genesis file (or any number that is small than the total number of pools that stays active in the local cluster.)

I think that's a good idea + changing this test to verify this. There are a lot of "smaller pools" on mainnet. So if we had nOpt = 2 (currently it is 3 I believe) then we would have, at least 1 "smaller pool" (as there are 3 pools in the integration tests). So, for this one pool the change of non-myopic rewards score should be visible in case of changing stake.